lib/auth/keygen/keygen.go

```diff
@@ -206,6 +206,9 @@ func (k *Keygen) GenerateUserCertWithoutValidation(c services.UserCertParams) ([
 	if c.Generation > 0 {
 		cert.Permissions.Extensions[teleport.CertExtensionGeneration] = fmt.Sprint(c.Generation)
 	}
+	if c.Bot {
+		cert.Permissions.Extensions[teleport.CertExtensionBot] = ""
```

Comment:
Maybe we should put the name of the Bot into the extension rather than just setting the extension - I figure it might be useful eventually?

Reply:
Good thought, I've converted the embedded cert field to a string containing the bot name (and renamed the field where relevant).